backend/service/audit/storage/sql/sql.go

```diff
@@ -132,17 +132,80 @@ func (c *client) UnsentEvents(ctx context.Context) ([]*auditv1.Event, error) {
 	return c.query(ctx, unsentEventsQuery)
 }
 
-func (c *client) ReadEvents(ctx context.Context, start time.Time, end *time.Time) ([]*auditv1.Event, error) {
-	const readEventsRangeStatement = `
+func (c *client) CountEvents(ctx context.Context, start time.Time, end *time.Time) (int64, error) {
+	countEventsRangeStatement := `
+		SELECT COUNT(*) FROM audit_events
+		WHERE occurred_at BETWEEN $1::timestamp AND $2::timestamp
+	`
+
+	endTime := time.Now()
+	if end != nil {
+		endTime = *end
+	}
+
+	var count int64
+	row := c.db.QueryRow(countEventsRangeStatement, start, endTime)
+	err := row.Scan(&count)
+	if err != nil {
+		return 0, err
+	}
+
+	return count, nil
+}
+
+func (c *client) ReadEvents(ctx context.Context, start time.Time, end *time.Time, options *storage.ReadOptions) ([]*auditv1.Event, error) {
+	readEventsRangeStatement := `
 		SELECT id, occurred_at, details FROM audit_events
 		WHERE occurred_at BETWEEN $1::timestamp AND $2::timestamp
 		ORDER BY id
 	`
 
-	if end == nil {
-		return c.query(ctx, readEventsRangeStatement, start, time.Now())
+	endTime := time.Now()
+	if end != nil {
+		endTime = *end
+	}
+
+	args := []interface{}{start, endTime}
+	if options != nil {
+		switch {
+		case options.Limit != 0 && options.Offset != 0:
+			readEventsRangeStatement = fmt.Sprintf(`
+			%s
+			LIMIT $3 OFFSET $4
+			`, readEventsRangeStatement)
+			args = append(args, options.Limit, options.Offset)
+		case options.Limit != 0:
+			readEventsRangeStatement = fmt.Sprintf(`
+			%s
+			LIMIT $3
+			`, readEventsRangeStatement)
+			args = append(args, options.Limit)
+		case options.Offset != 0:
+			readEventsRangeStatement = fmt.Sprintf(`
+			%s
+			OFFSET $4
+			`, readEventsRangeStatement)
```

Comment:
nit: Can we change these to: 
```suggestion
			readEventsRangeStatement = fmt.Sprintf(`%s OFFSET $4`, readEventsRangeStatement)
```

I'm not sure the spacing really helps with readability.